helm/polaris/templates/service.yaml

```diff
@@ -0,0 +1,42 @@
+{{/*
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+ 
+   http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+*/}}
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "polaris.fullname" . }}
+  labels:
+    {{- include "polaris.labels" . | nindent 4 }}
+  annotations:
+    {{- if .Values.service.annotations }}
+      {{- tpl (toYaml .Values.service.annotations) . | nindent 4 }}
+    {{- end }}
+spec:
+  type: {{ .Values.service.type }}
+  selector:
+    {{- include "polaris.selectorLabels" . | nindent 4 }}
+  ports:
+    {{- range $portName, $portNumber := .Values.service.ports }}
```

Comment:
Could be good to separate the metrics port from the main API port into 2 distinct services. But that can be done later.

Reply:
For sure beneficial such as network policies and traffic managements. However, I am not seeing much helm deployment separate metrics away from their main API (good practices for sure but most are not doing it). I will be more than happy to do it but here are couple things I would like to get feedback with:
1. Should we put ingress on top of metrics endpoint (I am assuming no here?)
2. Should we support diff service settings for diff endpoints (e.g. sessionAffinity, service type, sessionAffinity, etc.)

Reply:
> Should we put ingress on top of metrics endpoint (I am assuming no here?)

No, and that's the main reason why in Nessie we split the services in two. You typically want to expose the main port, but not the admin/mgmt one. So if you need to put an ingress in front of it, or turn the service into a LoadBalancer or NodePort type, the change would affect both ports and potentially expose the mgmt port as well.

> Should we support diff service settings for diff endpoints

Yes, you could have e.g. different session affinities, but more importantly, you can turn the mgmt service into a headless one.

Finally, you can also set `publishNotReadyAddresses: true`. This allows pods to be addressable even if they are not ready, which allows health checks to provide more insights as to why the pod is not healthy.

But I agree to look into this later on.

Reply:
Thanks. I can put those in the next PR after current one got merged.